code/modules/antagonists/cult/cult_structures.dm

```diff
@@ -12,16 +12,34 @@
 	var/cult_examine_tip
 	/// The cooldown for when items can be dispensed.
 	COOLDOWN_DECLARE(use_cooldown)
+	/// Assigned cult team, set when cultistism is checked.
+	var/datum/team/cult/cult_team
+
+/obj/structure/destructible/cult/on_constructed(mob/builder)
+	var/datum/antagonist/cult/cultist = builder.mind?.has_antag_datum(/datum/antagonist/cult, TRUE)
+	cult_team = cultist?.get_team()
+	return
+
+/obj/structure/destructible/cult/proc/is_cultist_check(mob/fool)
+
+	if(!IS_CULTIST(fool))
+		return FALSE
+
+	if(isnull(cult_team))
+		var/datum/antagonist/cult/cultist = fool.mind?.has_antag_datum(/datum/antagonist/cult, TRUE)
+		cult_team = cultist?.get_team()
```

Comment:
This proc is not pure (it has side effects, in updating the cult team var) despite the name implying it is. You should include documentation in its auto doc to mention this.